sycl/doc/design/spirv-extensions/SPV_INTEL_loop_dependence_annotations.asciidoc

```diff
@@ -0,0 +1,285 @@
+|====
+2+^.^| Decoration | Extra Operands	| Enabling Capabilities
+| 6205 | *{decoration_name}* +
+Can only be applied to *OpFunctionCall* and _Atomic_ instructions. Indicates the
+list of access groups that the decorated instruction belongs to. Operand is one
+or more literals that correspond to the access groups.
+| _literal_, _literal_, ... +
+_Access Group 1_, _Access Group 2_, ... | *{llvm_capability_name}*
+|====
+--
+
+
+
+=== Memory Operands
+
+Modify Section 3.26, "Memory Operands", adding these rows to the Memory Operand table:
+
+--
+[cols="^.^2,16,5",options="header"]
+|====
+2+^.^| Memory Operands | Enabling Capabilities
+| 0x40000 | *{mem_op_name}* +
+Followed by a number _N_ that indicates how many access groups this operation
+belongs to, and _N_ literals that correspond to the access groups. Indicates
+that this memory operation belongs to the specified access group(s).
+| *{llvm_capability_name}*
+|====
+--
+
+=== Loop Control
+
+Modify Section 3.23, "Loop Control", adding these rows to the Loop Control table:
+
+--
+[cols="^.^2,16,5",options="header"]
+|====
+2+^.^| Loop Control | Enabling Capabilities
+| 0x4000000 | *{pa_loop_control}* +
+Followed by a number _N_ >= 1 that indicates how many operands will follow, and
+_N_ _<id>_'s that are the result of *{aggregator_name}*. Indicates that for each
+list of access groups pointed to by an _<id>_, all operations with those access
+groups do not have any loop-carried data or memory-ordering dependencies carried by this loop.
+| *{llvm_capability_name}*
+| 0x8000000 | *{da_loop_control}* +
+Followed by a number _N_ >= 1 that indicates how many operands will follow, and
+_N_ pairs {_<id>_, _S_}. _<id>_ is a list of access groups coming from
+*{aggregator_name}*. _S_ is a literal >=0 indicating that the loop-carried
+dependence distance between any operations that belong to the specified access
+group(s) in _<id>_ is guaranteed to be greater than _S_. This means that there
+is no dependence with distance < _S_, but there could be a dependence with
+distance >= _S_. _S_=0 means that there are no loop-carried data dependencies
+carried by this loop between the operations.
+| *{sycl_capability_name}*
+|====
+--
+
+// == Issues
+
+// . Should we add an agregator instruction (e.g., *OpAGListDeclIntel*) that takes
+// the result of multiple *{instruction_name}* and generates a single <id> that
+// gets used by the decorations, memory operands, and loop controls?
+// +
+// --
+// *UNRESOLVED*: This may make the decorations, memory operands, and loop controls
+// simpler at the expense of adding an arbitrary number of instructions since an
+// agregator instruction will be needed for each location that uses access groups.
+// --
+
+== Revision History
+
+[cols="5,15,15,70"]
+[grid="rows"]
+[options="header"]
+|========================================
+|Rev|Date|Author|Changes
+|1|2023-04-17|Adel Ejjeh|*Initial revision*
+|2|2023-03-04|Adel Ejjeh|*Make AGs literals, update tokens*
```

Comment:
I'm guessing this should be 2024, otherwise the 2nd revision came before the first.

```suggestion
|2|2024-03-04|Adel Ejjeh|*Make AGs literals, update tokens*
```